amqp.go

```diff
@@ -26,7 +19,9 @@ func publishToAMQP(amqpURL string, exchange string, routingKey string, payload s
 		}).Info("try to publish to amqp")
 	}()
 
-	ch := getAMQPChannel(amqpURL)
+	ch, cleanUp := newAMQPChannel(amqpURL)
+	defer cleanUp()
+
```

Comment:
rather than dialing per every publish, can we retry on publish failures (and re-dial only on publish failures)?

Reply:
we can, but it requires some mutex locking for connection recreation and swapping, not sure if it worth the effort for low volume traffic mocking.

Also, having a new connection is not that expensive.

Reply:
fair enough